src/gui/win_nwg_ext/mod.rs

```diff
@@ -167,3 +170,67 @@ impl MenuItemEx for nwg::MenuItem {
         }
     }
 }
+
+pub trait WindowEx {
+    fn set_position_ex(&self, x: i32, y: i32);
+}
+pub fn dpi() -> i32 {
+    // prevents GDI DC resource leak
+    use winapi::um::wingdi::GetDeviceCaps;
+    use winapi::um::wingdi::LOGPIXELSX;
+    use winapi::um::winuser::{GetDC, ReleaseDC};
+    let screen = unsafe { GetDC(std::ptr::null_mut()) };
+    let dpi = unsafe { GetDeviceCaps(screen, LOGPIXELSX) };
+    let _ = unsafe { ReleaseDC(std::ptr::null_mut(), screen) };
+    dpi
+}
+pub fn logical_to_physical(x: i32, y: i32) -> (i32, i32) {
+    use muldiv::MulDiv;
+    use winapi::um::winuser::USER_DEFAULT_SCREEN_DPI;
+    let dpi = dpi();
+    let x = x.mul_div_round(dpi, USER_DEFAULT_SCREEN_DPI).unwrap_or(x);
+    let y = y.mul_div_round(dpi, USER_DEFAULT_SCREEN_DPI).unwrap_or(y);
+    (x, y)
+}
+/// # Safety
```

Comment:
Please put a proper safety comment here, i.e. under what conditions is it safe to call this function. Or if it's always safe to call, wrap the inner unsafe call with `unsafe` and do not mark this function as `unsafe`.

Reply:
I don't really know, this is a copy-paste from the NWG gui framework to fix a leak, and it doesn't document the safety requirements.

Think wrapping inner call should be fine, did it in some other examples from NWG, so will do that